anvil/anvil-compiler/src/main/java/com/duckduckgo/anvil/compiler/ContributeToActivityStarterCodeGenerator.kt

```diff
@@ -111,6 +122,39 @@ class ContributeToActivityStarterCodeGenerator : CodeGenerator {
                     )
                     .build(),
             )
+            .addFunction(
+                FunSpec.builder("map")
+                    .addModifiers(KModifier.OVERRIDE)
+                    .addParameter(
+                        "deeplinkActivityParams",
+                        deeplinkActivityParamsFqName.asClassName(module),
+                    )
+                    .returns(
+                        activityParamsFqName.asClassName(module).copy(nullable = true),
+                    )
+                    .addCode(
+                        """
+                            val screenName = deeplinkActivityParams.screenName.takeUnless { it.isEmpty() } ?: return null
+                            return if (screenName == %S) {
+                                if (deeplinkActivityParams.jsonArguments.isEmpty()) {
+                                    val instance = tryCreateObjectInstance(%T::class.java)
+                                    if (instance != null) {
+                                        return instance
+                                    }
+                                }
+                                tryCreateActivityParams(%T::class.java, deeplinkActivityParams)
+                            } else {
+                                null
+                            }
```

Comment:
When the screen doesn't have a name (ie. empty in `ContributeToActivityStarter`. This code generates something like
```kotlin
val screenName = ...
return if (screenName == "") {...} else {...}
```
The method eventually returns `null` because `val screenName = ...` returns null right in there, but still the following if statement is wrong.
I'd change to something like

```suggestion
                            val screenName = deeplinkActivityParams.screenName.takeUnless { it.isEmpty() } ?: return null
                            val definedScreenName = %S
                            if (definedScreenName.isNullOrBlank() return null
                            
                            return if (screenName == definedScreenName) {
                                if (deeplinkActivityParams.jsonArguments.isEmpty()) {
                                    val instance = tryCreateObjectInstance(%T::class.java)
                                    if (instance != null) {
                                        return instance
                                    }
                                }
                                tryCreateActivityParams(%T::class.java, deeplinkActivityParams)
                            } else {
                                null
                            }
```